apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs

```diff
@@ -453,67 +490,60 @@ where
-                    })
-                    .await
-                    .change_context(InsertionError)
-                    .attach_printable("Could not read parent data types")?
-                    .data_types,
-            )
-            .for_each(|data_type| {
-                ontology_type_resolver.add_open(Arc::new(data_type.schema));
+            .for_each(|parent| {
+                let parent_id = DataTypeId::from_url(&parent.schema.id);
+                if let Some(inheritance_data) = parent_inheritance_data.remove(&parent_id) {
+                    ontology_type_resolver.add_closed(
+                        parent_id,
+                        Arc::new(parent.schema),
+                        Arc::new(inheritance_data),
+                    );
+                } else {
+                    if !parent.schema.all_of.is_empty() {
+                        tracing::warn!("No inheritance data found for `{}`", parent.schema.id);
+                    }
+                    ontology_type_resolver.add_open(parent_id, Arc::new(parent.schema));
+                }
             });
 
-        for inserted_data_type in &inserted_data_types {
-            ontology_type_resolver.add_open(Arc::clone(inserted_data_type));
+        for (data_type_id, inserted_data_type) in &inserted_data_types {
+            ontology_type_resolver.add_open(*data_type_id, Arc::clone(inserted_data_type));
         }
         let schema_metadata = inserted_data_types
             .iter()
-            .map(|inserted_data_type| {
+            .map(|(data_type_id, _)| {
                 ontology_type_resolver
-                    .resolve_data_type_metadata(&inserted_data_type.id)
+                    .resolve_data_type_metadata(*data_type_id)
                     .change_context(InsertionError)
             })
             .collect::<Result<Vec<_>, _>>()?;
 
         let data_type_validator = DataTypeValidator;
-        for data_type in &inserted_data_types {
-            let closed_schema = data_type_validator
-                .validate(
-                    ontology_type_resolver
-                        .get_closed_data_type(&data_type.id)
-                        .change_context(InsertionError)?,
-                )
-                .await
-                .attach(StatusCode::InvalidArgument)
-                .change_context(InsertionError)?;
+        for (_, data_type) in &inserted_data_types {
+            // TODO: Validate ontology types on creation
+            //   see https://linear.app/hash/issue/H-2976/validate-ontology-types-on-creation
+            // let closed_schema = data_type_validator
+            //     .validate(
+            //         ontology_type_resolver
+            //             .get_closed_data_type(*data_type_id)
+            //             .change_context(InsertionError)?,
+            //     )
+            //     .await
+            //     .attach(StatusCode::InvalidArgument)
+            //     .change_context(InsertionError)?;
```

Comment:
This code did nothing before except pull all schemas from the cache. The validation basically was a no-op